plugins/processors/starlark/starlark.go

```diff
@@ -40,6 +40,41 @@ type Starlark struct {
 	applyFunc *starlark.Function
 	args      starlark.Tuple
 	results   []telegraf.Metric
+	state     State
+}
+
+// The struct that defines the shared state used
+type State struct {
+	content map[string]interface{}
+}
+
+// Load from the shared state, the value corresponding to the given key
+func (s *State) Load(key string) starlark.Value {
+	var value = s.content[key]
+	if value == nil {
+		return starlark.None
+	}
+	switch v := value.(type) {
+	case telegraf.Metric:
+		return WrapMetric(v)
+	default:
+		return v.(starlark.Value)
+	}
+}
+
+// Store the pair (key, value) into the shared state. If the value is None, the pair (key, value) will be
+// removed from the shared state if it exists
+func (s *State) Store(key string, value starlark.Value) {
```

Comment:
Should there be a configurable capacity limit here to prevent OOMs? 

I can foresee cases where the state store usage might blow up, for example if the name is defined based on tag values and the cardinality unexpectedly increases.

Reply:
that's a good question. I'm not sure if we should handle this or just add a warning to watch out for this.

Reply:
One alternative is to just the ability to check the size of the state and clear it if it exceeds a threshold. This could be implemented as additional built-in Starlark functions or just a configuration setting.

Running into an OOM is unlikely but for some of my use cases I need to trust that there is some reasonable guaranteed upper bound for memory usage when dealing with high-volume, high-cardinality data.

Reply:
I could see how it might be difficult to manage this in Starlark itself.. I'll give some thought to that.